lib/dal/src/action/runner.rs

```diff
@@ -170,7 +172,11 @@ impl From<ActionRunner> for ActionRunnerContentV1 {
             func_name: content.func_name,
             action_prototype_id: content.action_prototype_id,
             action_kind: content.action_kind,
-            resource: content.resource,
+            resource: content
+                .resource
+                .map(|r| serde_json::to_string(&r))
+                .transpose()
+                .expect("unable to serialize resource"),
```

Comment:
Are we okay with the panic here and below?

Reply:
yeah, we have a couple of those through the system, if its impossible to serialize the type its a bug and it will fail a test

serde serialization fallibility has a very narrow surface and sometimes its easier to let it panic if the type is changed than to pass through a "impossible" failure

but im ok with changing that, just took a shortcut that doesnt seem problematic